commands/create_kubeconfig.go

```diff
@@ -44,153 +89,205 @@ func init() {
+// createKubeconfigPreRunOutput shows our pre-check results
+func createKubeconfigPreRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
+	args := defaultCreateKubeconfigArguments()
+	err := verifyCreateKubeconfigPreconditions(args, cmdLineArgs)
+
+	if err != nil {
+		var headline = ""
+		var subtext = ""
+
+		switch {
+		case err.Error() == "":
+			return
+		case IsNotLoggedInError(err):
+			headline = "You are not logged in."
+			subtext = fmt.Sprintf("Use '%s login' to login or '--auth-token' to pass a valid auth token.", config.ProgramName)
+		case IsKubectlMissingError(err):
+			headline = "kubectl is not installed"
+			if runtime.GOOS == "darwin" {
+				subtext = "Please install via 'brew install kubernetes-cli' or visit\n"
+				subtext += fmt.Sprintf("%s for information on how to install kubectl", kubectlInstallURL)
+			} else if runtime.GOOS == "linux" {
+				subtext = fmt.Sprintf("Please visit %s for information on how to install kubectl", kubectlInstallURL)
+			} else if runtime.GOOS == "windows" {
+				subtext = fmt.Sprintf("Please visit %s to download a recent kubectl binary.", kubectlWindowsInstallURL)
+			}
+		default:
+			headline = err.Error()
 		}
-		return errors.New(errorMessage)
+
+		// print output
+		fmt.Println(color.RedString(headline))
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
 	}
 
-	endpoint := config.Config.ChooseEndpoint(cmdAPIEndpoint)
-	token := config.Config.ChooseToken(endpoint, cmdToken)
+}
 
-	if endpoint == "" {
+// verifyCreateKubeconfigPreconditions checks if all preconditions are met and
+// returns nil if yes, error if not
+func verifyCreateKubeconfigPreconditions(args createKubeconfigArguments, cmdLineArgs []string) error {
+	if config.Config.Token == "" && args.authToken == "" {
+		return microerror.Mask(notLoggedInError)
+	}
+	if args.apiEndpoint == "" {
 		return microerror.Mask(endpointMissingError)
 	}
-	if token == "" {
-		return errors.New("You are not logged in. Use '" + config.ProgramName + " login' to log in.")
+	if args.clusterID == "" {
+		return microerror.Mask(clusterIDMissingError)
 	}
-	if cmdClusterID == "" {
-		// use default cluster if possible
-		clusterID, _ := config.GetDefaultCluster(requestIDHeader, createKubeconfigActivityName, cmdLine, cmdAPIEndpoint)
-		if clusterID != "" {
-			cmdClusterID = clusterID
-		} else {
-			return errors.New("No cluster given. Please use the -c/--cluster flag to set a cluster ID.")
-		}
+
+	kubectlOkay := util.CheckKubectl()
+	if !kubectlOkay {
+		return microerror.Mask(kubectlMissingError)
 	}
+
 	return nil
 }
 
 // createKubeconfig adds configuration for kubectl
-func createKubeconfig(cmd *cobra.Command, args []string) {
-	endpoint := config.Config.ChooseEndpoint(cmdAPIEndpoint)
-	token := config.Config.ChooseToken(endpoint, cmdToken)
+func createKubeconfigRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
+	args := defaultCreateKubeconfigArguments()
+	result, err := createKubeconfig(args)
+
+	if err != nil {
```

Comment:
I would return here earlier

Reply:
The `os.Exit(1)` in that case replaces the `return`, after some error output.